doc/database/hierarchy.md

```diff
@@ -0,0 +1,241 @@
+    +int height
+    +int width
+  }
+  class Cultivar {
+    +int height
+    +int width
+  }
+
+  Variety <|-- Cultivar
+  Species <|-- Cultivar
+  Species <|-- Variety
+  Genus <|-- Species
+  Family <|-- Genus
+```
+
+Each of these entities can have their own attributes.
+E.g., a variety under a specie can have different height than the specie that it belongs to.
+
+## Unique Name
+
+In the database we have a column `unique_name`.
+The unique name are, either for abstract plants:
+
+- single latin word to specify "family", which always ends on: `-aceae`
+- single latin word to specify "genus"
+
+Or is built up by several words (for concrete plants):
+
+- single latin word to specify "genus"
+- single latin word to specify "specie"
+- optional single latin word to specify "variety"
+- optional several words in single quotes to specify "cultivar", which starts with a capital letter
+
+E.g. `Brassica oleracea italica 'Ramoso calabrese'`
+
+### Hybrid names
+
+Hybrid names are built up differently.
+Either two parent binomials, separated by a "x" or "×" or a given binomial, with or without an intercalated "×"
+(see [Wikipedia](https://en.wikipedia.org/wiki/Hybrid_name).
+So the name does not necessarily say if a plant is a hybrid.
+
+### Rendering
+
+The unique name must be rendered:
+
+- Latin name in *italics*.
+- Cultivar name normal font in single quotes (as in database).
+
+E.g.: *Brassica oleracea italica* 'Ramoso calabrese'
+
+### Rules
+
+We know about names (abstract and concrete, including hybrid):
+
+- If it contains more than one word, it is a concrete plant.
+- If it contains only one word, it is an abstract plant (family or genus).
+- All entries with 2 words are a specie (de: Art),
+- All entries with 3 or more words are either:
+  - variety (de: Varietät) if all is spelled italic, or
+  - are cultivar (de: Sorte) if last part of the name (can be more than 1 word!) is not latin, not italic, is in 'single quotes'
+- All entries with a single x between the words are hybrid.
+  We treat them like specie.
+
+## Attributes
+
+Plants are additionally classified as:
+
+- is_concrete_plant (is a concrete plant as opposite to an abstract plant)
+- is_tree (as search help within the tree layer)
+
+## Mappings
+
+### from Permapeople
+
+Column names should be:
+
+- `unique_name` (see above), instead of `binomial_name`
+- `common_name_en` (still with arrays of Text as content)
+- `common_name_de` should be from Wikipedia (still with arrays of Text as content), `german_name` gets removed
+- `hardiness_zone` (usda in documentation)
+- `light_requirement` (and not `sun`)
+- `water_requirement` (and not `water`)
+- `soil_texture` (and not `soil_type`)
+- `height` (and not `mature_size_height`, mature height in documentation)
+- `width` (and not `mature_size_width`, mature width in documentation)
+- `propagation_method` (and not `propagation`)
+- `growth_rate` (and not `growth`)
+- `has_drought_tolerance`
+- `id` should be `permaplant_id`
```

Comment:
They have an internal permapeople `id`, while we also have an id. Should we call it `permapeople_id` instead? 

Additionally, let's add a source column specifying the dataset i.e. `permapeople`, `practicalplants`,`reinsaat`